pkg/util/podhelper/wait.go

```diff
@@ -0,0 +1,121 @@
+package podhelper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/loft-sh/log"
+	"github.com/loft-sh/vcluster/pkg/util/clihelper"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+func WaitForReadyPod(ctx context.Context, kubeClient kubernetes.Interface, namespace, name, container string, log log.Logger) error {
+	now := time.Now()
+	err := wait.PollUntilContextTimeout(ctx, time.Second*2, time.Minute*2, true, func(ctx context.Context) (bool, error) {
+		pod, err := kubeClient.CoreV1().Pods(namespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			// this is a fatal
+			return false, fmt.Errorf("error trying to retrieve pod %s/%s: %w", namespace, name, err)
+		}
+
+		found := false
+		for _, containerStatus := range pod.Status.ContainerStatuses {
+			if containerStatus.State.Running != nil && containerStatus.Ready {
+				if containerStatus.Name == container {
+					found = true
+				}
+
+				continue
+			} else if containerStatus.State.Terminated != nil || (containerStatus.State.Waiting != nil && clihelper.CriticalStatus[containerStatus.State.Waiting.Reason]) {
+				// if the container is completed that is fine as well
+				if containerStatus.State.Terminated != nil && containerStatus.State.Terminated.ExitCode == 0 {
+					found = true
+					continue
+				}
+
+				reason := ""
+				message := ""
+				if containerStatus.State.Terminated != nil {
+					reason = containerStatus.State.Terminated.Reason
+					message = containerStatus.State.Terminated.Message
+				} else if containerStatus.State.Waiting != nil {
+					reason = containerStatus.State.Waiting.Reason
+					message = containerStatus.State.Waiting.Message
+				}
+
+				out, err := kubeClient.CoreV1().Pods(namespace).GetLogs(pod.Name, &corev1.PodLogOptions{
+					Container: container,
+				}).Do(context.Background()).Raw()
```

Comment:
Should this be `ctx` instead of `context.Background()`?  Not sure if the polling method cancels that context at an inopportune time messing up the log output.